### PR TITLE
Add Image comparison helpers

### PR DESCRIPTION
**Pull request by koglerch13**
## Change list

Based on appium/appium-base-driver#203
 
## Types of changes

What types of changes are you proposing/introducing to .NET client?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds value to the project)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Details

Add access to Appiums image comparison Apis (similar to https://github.com/appium/java-client/pull/870)

### CODE REVIEW

File: src/Appium.Net/Appium/AppiumDriver.cs [hunk 1]
```diff
@@ -534,6 +535,62 @@ public string StopRecordingScreen(IScreenRecordingOptions options)
...
+
+        public OccurenceMatchingResult FindImageOccurence(string base64Image1, string base64Image2, OccurenceMatchingOptions options = null)
+        {
+            var parameters = new Dictionary<string, object> {
+                { "mode", ComparisonMode.MatchTemplate },
+                { "firstImage", base64Image1 },
+                { "secondImage", base64Image2 }
+            };
+
+            if (options != null)
```
Comment (mykola-mokhnach) [inline src/Appium.Net/Appium/AppiumDriver.cs]: can we use [nullable](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/nullable-types/index) types notation?

Comment (mykola-mokhnach) [reply src/Appium.Net/Appium/AppiumDriver.cs]: same comment to other methods

File: src/Appium.Net/Appium/ImageComparison/ComparisonResult.cs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+using System.Drawing;
+using System.IO;
+
+namespace OpenQA.Selenium.Appium.ImageComparison
+{
+    public abstract class ComparisonResult
+    {
+        public string Visualization
+        {
+            get { return Result["visualization"].ToString(); }
```
Comment (mykola-mokhnach) [inline src/Appium.Net/Appium/ImageComparison/ComparisonResult.cs]: does C# have a built in wrapper type for png bitmaps?

File: src/Appium.Net/Appium/ImageComparison/ComparisonOptions.cs [hunk 3]
```diff
@@ -0,0 +1,11 @@
+﻿using System.Collections.Generic;
+
+namespace OpenQA.Selenium.Appium.ImageComparison
+{
+    public abstract class ComparisonOptions
+    {
+        public bool Visualize { get; set; }
```
Comment (mykola-mokhnach) [inline src/Appium.Net/Appium/ImageComparison/ComparisonOptions.cs]: some documentation on the available options would be very helpful

File: src/Appium.Net/Appium/ImageComparison/FeaturesMatchingOptions.cs [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    public class FeaturesMatchingOptions : ComparisonOptions
+    {
+        public string DetectorName { get; set; }
+        public string MatchFunc { get; set; }
+        public int GoodMatchesFactor { get; set; }
+
+        public override Dictionary<string, object> GetParameters()
+        {
+            return new Dictionary<string, object> {
+                { "detectorName", DetectorName },
```
Comment (mykola-mokhnach) [inline src/Appium.Net/Appium/ImageComparison/FeaturesMatchingOptions.cs]: an option should only be included into the request if its value has been explicitly changed

**Comment by mykola-mokhnach** [comment]
Can we have at least one functional test for the implemented APIs?

File: src/Appium.Net/Appium/AppiumDriver.cs (hunk 1 above)
Comment (koglerch13) [reply src/Appium.Net/Appium/AppiumDriver.cs]: I don't think so:
> T cannot be a reference type.
( from https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/nullable-types/index)

**Comment by akinsolb** [comment]
Hey @koglerch13 can you update the Changelog document? It’s the root of the project. The next release will be v4.1.0

**Comment by akinsolb** [comment]
Also can you pull upstream master into your branch?

File: src/Appium.Net/Appium/AppiumDriver.cs (hunk 1 above)
Comment (akinsolb) [reply src/Appium.Net/Appium/AppiumDriver.cs]: @mykola-mokhnach Did you mean:
`options?.parameters.Add("options", options.GetParameters());`

Comment (mykola-mokhnach) [reply src/Appium.Net/Appium/AppiumDriver.cs]: @akinsolb Yes, something like that to explicitly show that options is nullable

**Comment by koglerch13** [comment]
I just realized i messed something up w/ the git history. 
I'll probably create a new fork and open another pull request as soon as I have time.

**Pull request closed without merging**